ntoskrnl/mm/pagefile.c

```diff
@@ -89,6 +89,16 @@ static BOOLEAN MmSystemPageFileLocated = FALSE;
 
 /* FUNCTIONS *****************************************************************/
 
+static inline VOID UpdatePeakCommitment(VOID)
+{
+    // HACK: MmTotalCommittedPages should be adjusted consistently with
+    // other counters at different places.
+    MmTotalCommittedPages = MiMemoryConsumers[MC_SYSTEM].PagesUsed + MiMemoryConsumers[MC_USER].PagesUsed + MiUsedSwapPages;
```

Comment:
Since you duplicate the function, keep it in sync!

Reply:
Don't duplicate it, please. You can put a `static inline` function in a header file if needed.

Reply:
I used it once now with a ` static inline` declaration in a new mm.h but now it no longer compiles due to : 

```
In file included from ../ntoskrnl/mm/balance.c:18:
../ntoskrnl/mm/mm.h:2:20: error: 'UpdateTotalCommittedPages' used but never defined [-Werror]
 static inline VOID UpdateTotalCommittedPages(VOID);
                    ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

Could you help me please ?

Reply:
inline function `UpdateTotalCommittedPages` needs its definition. Where is the contents?

Reply:
defined in mm.h and implemented in balance.c

Reply:
(sorry i'm not familiar with `static inline`)

Reply:
Done